functions/src/functions/onUserWritten.ts

```diff
@@ -8,41 +8,30 @@
 
 import { type User, userConverter } from '@stanfordbdhg/engagehf-models'
 import { logger } from 'firebase-functions'
-import { onDocumentWritten } from 'firebase-functions/v2/firestore'
+import { onDocumentCreated } from 'firebase-functions/v2/firestore'
 import { DatabaseConverter } from '../services/database/databaseConverter.js'
 import { type Document } from '../services/database/databaseService.js'
 import { getServiceFactory } from '../services/factory/getServiceFactory.js'
 
-export const onUserWritten = onDocumentWritten(
+export const onUserWritten = onDocumentCreated(
   'users/{userId}',
   async (event) => {
+    if (event.data === undefined) return
     const factory = getServiceFactory()
     const userService = factory.user()
     try {
-      if (
-        event.data?.before.exists !== true &&
-        event.data?.after.exists === true
-      ) {
-        const converter = new DatabaseConverter(userConverter.value)
-        const userDoc: Document<User> = {
-          id: event.params.userId,
-          path: event.document,
-          content: converter.fromFirestore(event.data.after),
-          lastUpdate: new Date(event.time),
-        }
-        await userService.finishUserEnrollment(userDoc)
+      const converter = new DatabaseConverter(userConverter.value)
+      const userDoc: Document<User> = {
+        id: event.params.userId,
+        path: event.document,
+        content: converter.fromFirestore(event.data),
+        lastUpdate: new Date(event.time),
       }
+      await userService.finishUserEnrollment(userDoc)
     } catch (error) {
       logger.error(
         `Error finishing enrollment for user with id '${event.params.userId}' on change of user: ${String(error)}`,
       )
     }
-    try {
-      await userService.updateClaims(event.params.userId)
```

Comment:
As noted in our meeting; it would still make sense to update the claims here to allow token refreshes to get access to them while logged in.